docs: add datepicker isAbsolute/isRelative options (#1227)
apify/apify-docs: sources/platform/actors/development/actor_definition/input_schema/specification.md

![Apify Actor input schema - country input](./images/input-schema-country.png)

Example of date selection using absolute and relative `datepicker` editor:

```json
{
    "absolute_date": {
        "title": "Date",
        "type": "string",
        "description": "Select date",
        "editor": "datepicker"
    },
    "relative_date": {
        "title": "Relative date",
        "type": "string",
        "description": "Select date",
        "editor": "datepicker",
        "allowAbsolute": false,
        "allowRelative": true
    },
    "any_date": {
        "title": "Any date",
        "type": "string",
        "description": "Select date",
        "editor": "datepicker",
        "allowRelative": true
    }
}
```

> Rendered input for `absolute_date` property. See that by default, the date picker allows selecting only an absolute date in the form `YYYY-MM-DD`:

Reviewer: ```suggestion
The `absolute_date` property renders a date picker that allows selection of a specific date in the format `YYYY-MM-DD`.
```

Reviewer 2: Also, I'd stress here that this is the value returned by the field. See comment bellow.

![Apify Actor input schema - country input](./images/input-schema-date-absolute.png)

> Rendered input for `relative_date` property in form of `+/- {number} {unit}`. (The `allowAbsolute` is explicitly set to `false`):

Reviewer: ```suggestion
The `relative_date` property renders an input field that accepts dates in the `+/- {number} {unit}`. nthe `allowAbsolute` parameter is set to `false` to restrict input to relative dates only.
```

Reviewer 2: IMHO, to make this clear, we should write that 

> ...renders an input field that enables the user to choose the relative date and returns the value in the format `+/- {number} {unit}`, for example, `+ 2 days`.

This is a documentation for Actor developers so we need to focus on what is returned by the field which is what they work with.

![Apify Actor input schema - country input](./images/input-schema-date-relative.png)

> Rendered input for `any_date` property. The date picker allows selecting both absolute and relative dates, and it's up to the Actor author to parse recognize the chosen form:

Reviewer: ```suggestion
The `any_date` property renders a date picker that accepts both absolute and relative dates. The Actor author is  responsible for parsing and interpreting the selected date format.
```

| `allowAbsolute` | Boolean                                                                                                                                              | No                                         | Used only with `datepicker` editor for absolute date value picker in form `YYYY-MM-DD`. This defaults to `true`.                                                                                                                                                                                                                                                                                    |

> | `allowRelative` | Boolean                                                                                                                                              | No                                         | Used only with `datepicker` editor for relative input in form `+/- {number} {unit}`.<br/>The value passed to Actor input is in plain text  e.g. `"+ 3 weeks"` and it's up to Actor author to parse it.<br/>Supported units are: days, weeks, months and years.<br/>If only relative input is required, `allowAbsolute` needs to be explicitly set to `false`, otherwise both formats are supported. |

Author: I added regex that can be used to parse the relative value:

![Screenshot 2024-10-13 at 0 17 59](https://github.com/user-attachments/assets/debfe0e5-27ff-46e9-ae2f-6e92fdc038a3)

![Apify Actor input schema - country input](./images/input-schema-country.png)

Example of date selection using absolute and relative `datepicker` editor:

```json
{
    "absoluteDate": {
        "title": "Date",
        "type": "string",
        "description": "Select absolute date in format YYYY-MM-DD",
        "editor": "datepicker",
        "pattern": "^(\\d{4})-(0[1-9]|1[0-2])-(0[1-9]|[12]\\d|3[01])$"
    },
    "relativeDate": {
        "title": "Relative date",
        "type": "string",
        "description": "Select relative date in format +/- {number} {unit}",
        "editor": "datepicker",
        "dateType": "relative",
        "pattern": "^([+-])\\s*(\\d+)\\s*(day|week|month|year)s?$"
    },
    "anyDate": {
        "title": "Any date",
        "type": "string",
        "description": "Select date in format YYYY-MM-DD or +/- {number} {unit}",
        "editor": "datepicker",
        "dateType": "absoluteOrRelative",
        "pattern": "^(\\d{4})-(0[1-9]|1[0-2])-(0[1-9]|[12]\\d|3[01])$|^([+-])\\s*(\\d+)\\s*(day|week|month|year)s?$"
    }
}
```

> The `absoluteDate` property renders a date picker that allows selection of a specific date and returns string value in the format `YYYY-MM-DD`. Validation is ensured thanks to `pattern` field. In this case the `dateType` property is omitted, as it defaults to `"absolute"`.

Reviewer: ```suggestion
The `absoluteDate` property renders a date picker that allows selection of a specific date and returns string value in `YYYY-MM-DD` format. Validation is ensured thanks to `pattern` field. In this case the `dateType` property is omitted, as it defaults to `"absolute"`.
```

![Apify Actor input schema - country input](./images/input-schema-date-absolute.png)

> The `relativeDate` property renders an input field that enables the user to choose the relative date and returns the value in the format `+/- {number} {unit}`, for example `"+ 2 days"`. The `dateType` parameter is set to `"relative"` to restrict input to relative dates only.

Reviewer: ```suggestion
The `relativeDate` property renders an input field that enables the user to choose the relative date and returns the value in `+/- {number} {unit}` format, for example `"+ 2 days"`. The `dateType` parameter is set to `"relative"` to restrict input to relative dates only.
```

> | `dateType`      | One of <ul><li>`absolute`</li><li>`relative`</li><li>`absoluteOrRelative`</li></ul>                                                                  | No                                         | This property, which is only available with `datepicker` editor, specifies what date format should visual editor accept (There is no validation and in JSON editor it accept any string).<br/><br/><ul><li>`absolute` value enables date input in `YYYY-MM-DD` format. To parse returned string regex like this can be used: `^(\d{4})-(0[1-9]\|1[0-2])-(0[1-9]\|[12]\d\|3[01])$`.</li><br/><li>`relative` value enables relative date input in <br/>`+/- {number} {unit}` format. <br/>Supported units are: days, weeks, months, years.<br/><br/>The input is passed to the Actor as plain text (e.g., "+3 weeks"). To parse it, regex like this can be used: `^([+-])\s*(\d+)\s*(day\|week\|month\|year)s?$`.</li><br/><li>`absoluteOrRelative` value enables both absolute and relative formats and user can switch between them. It's up to Actor author to parse a determine actual used format - regexes above can be used to check whether the returned string match one of them.</li></ul><br/>Defaults to `absolute`. |

Reviewer: ```suggestion
| `dateType`      | One of <ul><li>`absolute`</li><li>`relative`</li><li>`absoluteOrRelative`</li></ul>                                                                  | No                                         | This property, which is only available with `datepicker` editor, specifies what date format should visual editor accept (The JSON editor accepts any string without validation.).<br/><br/><ul><li>`absolute` value enables date input in `YYYY-MM-DD` format. To parse returned string regex like this can be used: `^(\d{4})-(0[1-9]\|1[0-2])-(0[1-9]\|[12]\d\|3[01])$`.</li><br/><li>`relative` value enables relative date input in <br/>`+/- {number} {unit}` format. <br/>Supported units are: days, weeks, months, years.<br/><br/>The input is passed to the Actor as plain text (e.g., "+3 weeks"). To parse it, regex like this can be used: `^([+-])\s*(\d+)\s*(day\|week\|month\|year)s?$`.</li><br/><li>`absoluteOrRelative` value enables both absolute and relative formats and user can switch between them. It's up to Actor author to parse a determine actual used format - regexes above can be used to check whether the returned string match one of them.</li></ul><br/>Defaults to `absolute`. |
```

| `isSecret`   | Boolean                                                                                                                                                                                                               | No                                         | Specifies whether the input field<br />will be stored encrypted.<br />Only available <br />with `textfield` and `textarea` editors. |
| `isAbsolute` | Boolean                                                                                                                                                                                                               | No                                         | Used with `datepicker` editor for absolute date value picker in form `YYYY-MM-DD`. This defaults to `true`.                          |

> | `isRelative` | Boolean                                                                                                                                                                                                               | No                                         | Used with `datepicker` editor for relative input in form `+/- {number} {unit}` e.g. `+ 3 weeks`.                                    |

Reviewer 3: They are missing what units are allowed.

Author: Done, added supported units and also mention that `isAbsolute` and `isRelative` are also supported together

> | `allowAbsolute` | Boolean                                                                                                                                                                                                               | No                                         | Used only with `datepicker` editor for absolute date value picker in form `YYYY-MM-DD`. This defaults to `true`.                                                                                                                                                                               |

Reviewer 2: The properties are fine, but I don't understand what they mean well. Perhaps there should be better explanation. Is it:
- If I enable allow absolute then I am able to pick certain date
- If I enable allow relative then I can type "3 months"??
But what is the returned value is what? Always a date in ISO format?

Reviewer 4: I think the `allowRelative` explanation is good, but the `allowAbsolute` can be improved. Also could maybe show an example like we do it for other more complicated fields.

Reviewer 2: So, in this case, we should have 2 different editors or "resources types". You should not have varying outputs from a single component.

Reviewer 2: Like one function should always return either date object or relative string but not both.

Reviewer 4: But the main requirement from store team is to have a single field

Reviewer 2: Ok, that makes sense then.
